// @notifi-network/notifi-axios-adapter/dist/NotifiAxiosService.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotifiAxiosService = void 0;
const axios_1 = __importDefault(require("axios"));
const beginLogInByTransaction_1 = __importDefault(require("./mutations/beginLogInByTransaction"));
const broadcastMessageImpl_1 = __importDefault(require("./mutations/broadcastMessageImpl"));
const completeLogInByTransaction_1 = __importDefault(require("./mutations/completeLogInByTransaction"));
const connectWalletImpl_1 = __importDefault(require("./mutations/connectWalletImpl"));
const createAlertImpl_1 = __importDefault(require("./mutations/createAlertImpl"));
const createDiscordTargetImpl_1 = __importDefault(require("./mutations/createDiscordTargetImpl"));
const createEmailTargetImpl_1 = __importDefault(require("./mutations/createEmailTargetImpl"));
const createSmsTargetImpl_1 = __importDefault(require("./mutations/createSmsTargetImpl"));
const createSourceGroupImpl_1 = __importDefault(require("./mutations/createSourceGroupImpl"));
const createSourceImpl_1 = __importDefault(require("./mutations/createSourceImpl"));
const createSupportConversationImpl_1 = __importDefault(require("./mutations/createSupportConversationImpl"));
const createTargetGroupImpl_1 = __importDefault(require("./mutations/createTargetGroupImpl"));
const createTelegramTargetImpl_1 = __importDefault(require("./mutations/createTelegramTargetImpl"));
const createWebhookTargetImpl_1 = __importDefault(require("./mutations/createWebhookTargetImpl"));
const deleteAlertImpl_1 = __importDefault(require("./mutations/deleteAlertImpl"));
const deleteSourceGroupImpl_1 = __importDefault(require("./mutations/deleteSourceGroupImpl"));
const deleteTargetGroupImpl_1 = __importDefault(require("./mutations/deleteTargetGroupImpl"));
const logInFromDappImpl_1 = __importDefault(require("./mutations/logInFromDappImpl"));
const markFusionNotificationHistoryAsRead_1 = __importDefault(require("./mutations/markFusionNotificationHistoryAsRead"));
const refreshAuthorizationImpl_1 = __importDefault(require("./mutations/refreshAuthorizationImpl"));
const sendConversationMessagesImpl_1 = __importDefault(require("./mutations/sendConversationMessagesImpl"));
const sendEmailTargetVerificationRequestImpl_1 = __importDefault(require("./mutations/sendEmailTargetVerificationRequestImpl"));
const updateSourceGroupImpl_1 = __importDefault(require("./mutations/updateSourceGroupImpl"));
const updateTargetGroupImpl_1 = __importDefault(require("./mutations/updateTargetGroupImpl"));
const findTenantConfigImpl_1 = __importDefault(require("./queries/findTenantConfigImpl"));
const getAlertsImpl_1 = __importDefault(require("./queries/getAlertsImpl"));
const getConfigurationForDappImpl_1 = __importDefault(require("./queries/getConfigurationForDappImpl"));
const getConnectedWalletsImpl_1 = __importDefault(require("./queries/getConnectedWalletsImpl"));
const getConversationMessagesImpl_1 = __importDefault(require("./queries/getConversationMessagesImpl"));
const getDiscordTargetsImpl_1 = __importDefault(require("./queries/getDiscordTargetsImpl"));
const getEmailTargetsImpl_1 = __importDefault(require("./queries/getEmailTargetsImpl"));
const getFiltersImpl_1 = __importDefault(require("./queries/getFiltersImpl"));
const getFusionNotificationHistoryImpl_1 = __importDefault(require("./queries/getFusionNotificationHistoryImpl"));
const getNotificationHistoryImpl_1 = __importDefault(require("./queries/getNotificationHistoryImpl"));
const getSmsTargetsImpl_1 = __importDefault(require("./queries/getSmsTargetsImpl"));
const getSourceGroupsImpl_1 = __importDefault(require("./queries/getSourceGroupsImpl"));
const getSourcesImpl_1 = __importDefault(require("./queries/getSourcesImpl"));
const getTargetGroupsImpl_1 = __importDefault(require("./queries/getTargetGroupsImpl"));
const getTelegramTargetsImpl_1 = __importDefault(require("./queries/getTelegramTargetsImpl"));
const getTopicsImpl_1 = __importDefault(require("./queries/getTopicsImpl"));
const getUnreadNotificationHistoryCountImpl_1 = __importDefault(require("./queries/getUnreadNotificationHistoryCountImpl"));
const getWebhookTargetsImpl_1 = __importDefault(require("./queries/getWebhookTargetsImpl"));
class NotifiAxiosService {
    constructor(c) {
        this.jwt = null;
        this.setJwt = (jwt) => {
            this.jwt = jwt;
        };
        const a = axios_1.default.create({
            baseURL: c.gqlUrl,
        });
        a.interceptors.request.use((config) => {
            const jwt = this.jwt;
            if (jwt !== null) {
                return Object.assign(Object.assign({}, config), { headers: Object.assign(Object.assign({}, config.headers), { Authorization: `Bearer ${jwt}` }) });
            }
            return config;
        });
        this.beginLogInByTransaction = beginLogInByTransaction_1.default.bind(null, a);
        this.broadcastMessage = broadcastMessageImpl_1.default.bind(null, a);
        this.completeLogInByTransaction = completeLogInByTransaction_1.default.bind(null, a);
        this.connectWallet = connectWalletImpl_1.default.bind(null, a);
        this.createAlert = createAlertImpl_1.default.bind(null, a);
        this.createEmailTarget = createEmailTargetImpl_1.default.bind(null, a);
        this.createSmsTarget = createSmsTargetImpl_1.default.bind(null, a);
        this.createTargetGroup = createTargetGroupImpl_1.default.bind(null, a);
        this.createTelegramTarget = createTelegramTargetImpl_1.default.bind(null, a);
        this.createDiscordTarget = createDiscordTargetImpl_1.default.bind(null, a);
        this.createSource = createSourceImpl_1.default.bind(null, a);
        this.createSourceGroup = createSourceGroupImpl_1.default.bind(null, a);
        this.createSupportConversation = createSupportConversationImpl_1.default.bind(null, a);
        this.createWebhookTarget = createWebhookTargetImpl_1.default.bind(null, a);
        this.deleteAlert = deleteAlertImpl_1.default.bind(null, a);
        this.deleteSourceGroup = deleteSourceGroupImpl_1.default.bind(null, a);
        this.deleteTargetGroup = deleteTargetGroupImpl_1.default.bind(null, a);
        this.findTenantConfig = findTenantConfigImpl_1.default.bind(null, a);
        this.getAlerts = getAlertsImpl_1.default.bind(null, a);
        this.getConfigurationForDapp = getConfigurationForDappImpl_1.default.bind(null, a);
        this.getConnectedWallets = getConnectedWalletsImpl_1.default.bind(null, a);
        this.getConversationMessages = getConversationMessagesImpl_1.default.bind(null, a);
        this.getEmailTargets = getEmailTargetsImpl_1.default.bind(null, a);
        this.getFilters = getFiltersImpl_1.default.bind(null, a);
        this.getFusionNotificationHistory = getFusionNotificationHistoryImpl_1.default.bind(null, a);
        this.getNotificationHistory = getNotificationHistoryImpl_1.default.bind(null, a);
        this.getSmsTargets = getSmsTargetsImpl_1.default.bind(null, a);
        this.getSourceGroups = getSourceGroupsImpl_1.default.bind(null, a);
        this.getSources = getSourcesImpl_1.default.bind(null, a);
        this.getTargetGroups = getTargetGroupsImpl_1.default.bind(null, a);
        this.getTelegramTargets = getTelegramTargetsImpl_1.default.bind(null, a);
        this.getDiscordTargets = getDiscordTargetsImpl_1.default.bind(null, a);
        this.getTopics = getTopicsImpl_1.default.bind(null, a);
        this.getUnreadNotificationHistoryCount =
            getUnreadNotificationHistoryCountImpl_1.default.bind(null, a);
        this.getWebhookTargets = getWebhookTargetsImpl_1.default.bind(null, a);
        this.logInFromDapp = logInFromDappImpl_1.default.bind(null, a);
        this.refreshAuthorization = refreshAuthorizationImpl_1.default.bind(null, a);
        this.sendEmailTargetVerificationRequest =
            sendEmailTargetVerificationRequestImpl_1.default.bind(null, a);
        this.sendConversationMessages = sendConversationMessagesImpl_1.default.bind(null, a);
        this.updateSourceGroup = updateSourceGroupImpl_1.default.bind(null, a);
        this.updateTargetGroup = updateTargetGroupImpl_1.default.bind(null, a);
        this.markFusionNotificationHistoryAsRead =
            markFusionNotificationHistoryAsRead_1.default.bind(null, a);
    }
}
exports.NotifiAxiosService = NotifiAxiosService;
